activation/atxdb.go

```diff
@@ -157,40 +151,26 @@ func (db *ActivationDb) CalcActiveSetFromView(a *types.ActivationTx) (uint32, er
 //   NodeID, SequenceNumber, PrevATXID, LayerID, StartTick, PositioningATX.
 // - ATX LayerID is NipstLayerTime or more after the PositioningATX LayerID.
 // - The ATX view of the previous epoch contains ActiveSetSize activations
-func (db *ActivationDb) ValidateAtx(atx *types.ActivationTx) error {
+func (db *ActivationDb) SyntacticallyValidateAtx(atx *types.ActivationTx) error {
 	if atx.PrevATXId != *types.EmptyAtxId {
 		prevATX, err := db.GetAtx(atx.PrevATXId)
 		if err != nil {
 			return fmt.Errorf("validation failed: prevATX not found: %v", err)
 		}
+		if prevATX.NodeId.Key != atx.NodeId.Key {
+			return fmt.Errorf("previous ATX belongs to different miner. atx.Id: %v, atx.NodeId: %v, prevAtx.NodeId: %v",
+				atx.ShortId(), atx.NodeId.Key, prevATX.NodeId.Key)
+		}
 		if !prevATX.Valid {
 			return fmt.Errorf("validation failed: prevATX not valid")
 		}
 		if prevATX.Sequence+1 != atx.Sequence {
 			return fmt.Errorf("sequence number is not one more than prev sequence number")
 		}
-
-		prevAtxIds, err := db.GetNodeAtxIds(atx.NodeId)
-		if len(prevAtxIds) > 0 {
-			lastAtx := prevAtxIds[len(prevAtxIds)-1]
-			// last atx is not the one referenced
-			if lastAtx != atx.PrevATXId {
-				return fmt.Errorf("last atx is not the one referenced")
-			}
-		}
-
 	} else {
```

Comment:
If this is the first ATX of this miner, then they should provide the PoST proof for the initialization phase (should also be apart of the PoET challenge). Do we take care of it somewhere?

Reply:
https://github.com/spacemeshos/go-spacemesh/issues/1075